### PR TITLE
chore: restructure pallet part 2, workspace manifest file

### PR DESCRIPTION
**Pull request by neutrinoks**
- The original crate `pallet-move` is now moved to subdirectory `pallet`
- The RPC is now in subdirectory `rpc`
- The pallet's runtime API is moved into the pallet directly
- Added a workspace manifest file and centralized dependencies
- All related definitions and re-exports related to the former runtime-API are now within `pallet_move::api`

### CODE REVIEW

**Review by Rqnsom — APPROVED** [review]
LGTM, but let's wait for @asmie's approval as well

**Review by asmie — APPROVED** [review]
LGTM!

**Pull request merged**
